From 9d10c9744d0bdbb021e20e2cc8a1ceb0ef957946 Mon Sep 17 00:00:00 2001
From: Benjamin Tissoires <benjamin.tissoires@redhat.com>
Date: Thu, 18 Dec 2014 10:25:17 -0500
Subject: [PATCH 1/2] Fix Lenovo series key mapping

Signed-off-by: Benjamin Tissoires <benjamin.tissoires@redhat.com>
---
 .../30-keymap-module-thinkpad-acpi.fdi             |    7 ++++++-
 1 files changed, 6 insertions(+), 1 deletions(-)

diff --git a/fdi/information/10freedesktop/30-keymap-module-thinkpad-acpi.fdi b/fdi/information/10freedesktop/30-keymap-module-thinkpad-acpi.fdi
index b60b13e..ea7746c 100644
--- a/fdi/information/10freedesktop/30-keymap-module-thinkpad-acpi.fdi
+++ b/fdi/information/10freedesktop/30-keymap-module-thinkpad-acpi.fdi
@@ -30,7 +30,7 @@
         <append key="input.keymap.data" type="strlist">0x03:sleep</append> <!-- Fn+F4 suspend -->
         <append key="input.keymap.data" type="strlist">0x04:wlan</append> <!-- Fn+F5 wifi -->
         <append key="input.keymap.data" type="strlist">0x06:switchvideomode</append> <!-- Fn+F7 switch video output device -->
-        <append key="input.keymap.data" type="strlist">0x07:f22</append> <!-- Fn+F8 touchpadtoggle -->
+        <append key="input.keymap.data" type="strlist">0x07:f21</append> <!-- Fn+F8 touchpadtoggle -->
         <append key="input.keymap.data" type="strlist">0x08:f24</append> <!-- Fn+F9 undock -->
         <append key="input.keymap.data" type="strlist">0x0b:suspend</append> <!-- Fn+F12 hibernate -->
         <append key="input.keymap.data" type="strlist">0x0f:brightnessup</append> <!-- Fn+Home -->
@@ -41,6 +41,11 @@
 	<append key="input.keymap.data" type="strlist">0x15:volumedown</append> <!-- Volume Down button -->
         <append key="input.keymap.data" type="strlist">0x16:mute</append> <!-- Mute button -->
         <append key="input.keymap.data" type="strlist">0x17:prog1</append> <!-- ThinkPad/ThinkVantage button (high keycode: "vendor") -->
+	<append key="input.keymap.data" type="strlist">0x1a:f20</append> <!-- Mic Mute -->
+	<append key="input.keymap.data" type="strlist">0x1c:config</append> <!-- Fn+F9 Config -->
+	<append key="input.keymap.data" type="strlist">0x1d:search</append> <!-- Fn+F10 Search -->
+	<append key="input.keymap.data" type="strlist">0x1e:scale</append> <!-- Fn+F11 Scale -->
+	<append key="input.keymap.data" type="strlist">0x1f:file</append> <!-- Fn+F12 File -->
         <append key="info.capabilities" type="strlist">input.keymap</append>
       </match>
     </match>
-- 
1.7.1

